kazuki-toolkit/src/main/kotlin/com/anaplan/engineering/kazuki/toolkit/iso8601/Dtg.kt

```diff
@@ -35,203 +37,138 @@ class DtgProperties(private val dtg: Dtg) {
 
 class DtgFunctions(private val dtg: Dtg) {
 
-    val addDuration: (Duration) -> Dtg = function(
-        command = { duration -> dtg.properties.durationSinceFirstDtg.functions.addDuration(duration).properties.dtgAfterFirstDtg },
-        post = { duration, result -> result.functions.subtractDuration(duration) == dtg }
+    private val localDateTime by lazy { dtg.toLocalDateTime() }
+
+    val isEarlierThan = function(
+        command = { other: Dtg -> localDateTime < other.toLocalDateTime() },
+        post = { other, result ->
+            result iff (dtg.date.functions.isEarlierThan(other.date) || (
+                    dtg.date == other.date && dtg.time.functions.isEarlierThan(other.time)))
+        }
+    )
+
+    val addDuration = function(
+        command = { duration: Duration ->
+            dtg.toLocalDateTime().plusNanos(duration.milliseconds.toLong() * 1_000_000).toDtg()
```

Comment:
❗ Weird that there's no `plusMillis`...

Reply:
no micro or pico either.. :smile:  I suspect this is just due to the evolution of CPU time granularity